test/wolff_supervised_tests.erl

```diff
@@ -485,18 +464,18 @@ kafka_topic_cmd_base(Topic) ->
     " --zookeeper zookeeper:2181" ++
     " --topic '" ++ Topic ++ "'".
 
-assert_eq_optional_tail(Fun, ExpectedList) ->
-    ExpectedListMinusLast = lists:sublist(ExpectedList, length(ExpectedList) - 1),
-    case Fun() of
-        ExpectedList ->
-            ok;
-        ExpectedListMinusLast ->
-            %% sometimes the terminate callback seems to not have time
-            %% to insert into the ets, making this flaky.
-            ok;
-        Xs0 ->
-            ct:fail("unexpected result: ~p", [Xs0])
-    end.
-
 get_telemetry_seq(Table, Event) ->
     wolff_tests:get_telemetry_seq(Table, Event).
+
+assert_eventually_zero(Key, Tab) ->
+  case get_telemetry_seq(Tab, [wolff, Key]) of
+    [] ->
+      throw(no_event_recorded);
+    L ->
+      case lists:last(L) of
+        0 ->
+          ok;
+        X ->
+          throw({unexpected_eventual_state, X})
+      end
+  end.
```

Comment:
From the name of this function I would have expected it to wait a bit and check again if the counter is zero, if it was not zero in the first check but this does not seem to be the case. Is assert_eventually_zero a good name for this function?

Reply:
assert_last_event_is_zero ?